packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -396,6 +400,25 @@ export class DataStores implements IDisposable {
                 return context.attachState === AttachState.Attached;
             }).map(async ([contextId, context]) => {
                 const contextSummary = await context.summarize(fullTree, trackState);
+                const summary = contextSummary.summary;
+                assert(summary.type === SummaryType.Tree, "context summary is not a tree");
```

Comment:
This assert shouldn't be here, because it will throw if the datastore is unchanged; it will be a handle, not a tree.